cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/loadbalancer/AwsLoadBalancerMetadata.java

```diff
@@ -0,0 +1,40 @@
+package com.sequenceiq.cloudbreak.cloud.model.loadbalancer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * AWS specific extension of LoadBalancerMetadataBase. For AWS, the resource information we need is the ARN
+ * for the load balancer, and information about any target groups and listeners attached to the load
+ * balancer. This information is represented in the AwsTargetGroupMetadata object. Using the provided information,
+ * a user can use the AWS console and CLI to look up additional information about the load balancer as needed.
+ */
+public class AwsLoadBalancerMetadata extends LoadBalancerMetadataBase {
```

Comment:
this doesn't seem right to have provider specific stuff in the cloud-api modul. the purpose of this is to provide an abstraction layer

Reply:
Then do you know what module it should go in? I understand the purpose of the CB model is abstraction. But the load balancers are different enough between cloud providers that we couldn't find a way to complete abstract it out. We need this information to return anything useful about the load balancers, but I'm happy to move it if it belongs in a different part of the code.

Reply:
I would expect something similar what we have for credentials. Those are pretty different per providers.
@biharitomi @doktoric do you have any better suggestion/idea?

Reply:
Do you mean the `AwsCredentialPrerequisites`, `AzureCredentialPrerequisites`, and `GcpCredentialPrerequisites` classes? Because those are also in the cloud-api module.

Reply:
no, I meant `CloudCredential` which is kinda uniqe for the providers. I'm not sure why the prerequisite check generalized.
I'm asking the guys for input, because I think they might have better knowledge about the cloud modules. Let me ping them

Reply:
LoadBalancerMetadata can be an extends of DynamicModel and then you can use the parameter map for storing cloud provider related data. You can create a view object where you want to use the map elements e.g AwsCredentialView. 
In this way you can store the information in a class which is abstract enough.

Reply:
I would also prefer the approach which would not contain any provider specific info in the `API` module. 
I think the credential prerequisite stuff is there by a mistake and probably didn't catch during the review process.

Reply:
I will look into that. I admit to being frustrated that this review has been up for three weeks, and I'm only now getting feedback to significantly rework it.

Reply:
I understand the frustration, but I got to you in ~5 workdays after I got the email notification from Github. I know it's still a lot.

Reply:
Sorry, but we needed to finish the demoable implementation of an immediate feature before I could allocate a ~2 hours to read through a ~50 files PR.

Reply:
Yeah, the fact that I changed the package of one of the classes meant a lot of files that weren't otherwise changed ended up with different imports. I've actually undone that move, so there are less files touched now.

I've made `CloudLoadBalancerMetadata` extend `DynamicModel`, and created an `AwsLoadBalancerMetadataView` to parse the parameters. Everything still converts as expected according to the unit tests. Let me know how it looks.